e2e/util/validation.go

```diff
@@ -76,30 +79,48 @@ func GetRamenNameSpace(k8sClient *kubernetes.Clientset) (string, error) {
 
 // IsOpenShiftCluster checks if the given Kubernetes cluster is an OpenShift cluster.
 // It returns true if the cluster is OpenShift, false otherwise, along with any error encountered.
-func IsOpenShiftCluster(k8sClient *kubernetes.Clientset) (bool, error) {
-	discoveryClient := k8sClient.Discovery()
+func IsOpenShiftCluster(k8sClient client.Client) (bool, error) {
+	configList := &unstructured.Unstructured{}
+	configList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "config.openshift.io",
+		Version: "v1",
+		Kind:    "ClusterVersion",
+	})
 
-	apiGroups, err := discoveryClient.ServerGroups()
-	if err != nil {
-		return false, err
+	err := k8sClient.List(context.TODO(), configList)
+	if err == nil {
+		// found OpenShift only resource type, it is OpenShift
+		return true, nil
 	}
 
-	for _, group := range apiGroups.Groups {
-		if group.Name == "route.openshift.io" {
-			return true, nil
-		}
+	if meta.IsNoMatchError(err) {
+		// api server says no match for OpenShift only resource type,
+		// it is not OpenShift
```

Comment:
Not clear what this means. Don't we get NotFound error if the resource does not exist?

Reply:
No, if the GVK is not found then we get a NoMatchError. If the GVK exists but a CR of that name doesn't then we get a NotFound error.